### PR TITLE
Allow identifier to be PatientID or PatientName

### PR DESCRIPTION
**Pull request by spaiva**
(Previously from PR #145) 
- Created `$identifier` variable to allow parsing for **PatientID** or **PatientName**
- With this `$identifier` variable, we no longer need `$lookupCenterNameUsing`
- Regular Pattern (`$PatientID_regex`) of Patient ID is configurable on ProfileTemplate. 
- Removed duplicate method `getSubjectIDs()` from Profile Template, using the one from MRI.pm instead, as it is more robust. 
- Dependencies: Dicom-Archive [PR#38](https://github.com/aces/dicom-archive-tools/pull/38)

### CODE REVIEW

File: uploadNeuroDB/NeuroDB/MRI.pm [hunk 1]
```diff
@@ -975,30 +954,30 @@ sub createNewCandID {
...
     }
 
-    ###########################################################################	
+    ###########################################################################
     ###Get the centerID using the PSCID########################################
     ###########################################################################
 
     #extract the PSCID from $patientName
-    $subjectIDsref = getSubjectIDs($patientName,null,$dbhr);
+    $subjectIDsref = &Settings::getSubjectIDs($identifier,null,$dbhr);
```
Comment (cmadjar) [inline uploadNeuroDB/NeuroDB/MRI.pm]: I am not sure to follow. The function getSubjectIDs is removed from the profile but still called from the profile here?

Comment (cmadjar) [inline uploadNeuroDB/NeuroDB/MRI.pm]: Forget it... I just realize I was not looking at the profile template.. Haha. All good :)

File: uploadNeuroDB/NeuroDB/MRI.pm [hunk 2]
```diff
@@ -50,35 +50,14 @@ $VERSION = 0.2;
...
-	$subjectID{'CandID'} = my_trim(getScannerCandID($scannerID, $dbhr));
-	$subjectID{'visitLabel'} = my_trim($patientName);
-# subject data       	
-# old versions of this
-# qnts /([A-Z-]{3,4}\s+\d+)_(\d+)_([^_ ]+)/) or nihpd =~ /(\w{3}\d+)_(\d+)_([^_ ]+)/)
-    } elsif ($patientName =~ /([^_]+)_(\d+)_([^_ ]+)/) {
-	$subjectID{'PSCID'} = my_trim($1);
-	$subjectID{'CandID'} = my_trim($2);
-	$subjectID{'visitLabel'} = my_trim($3);
-	if(!subjectIDIsValid($subjectID{'CandID'}, $subjectID{'PSCID'}, $subjectID{'visitLabel'}, $dbhr)) {
```
Comment (MounaSafiHarab) [inline uploadNeuroDB/NeuroDB/MRI.pm]: @spaiva, @cmadjar:
We just need to be careful that this duplicate getSubjectID() function that got removed did not also remove some additional functionalities... for example subjectIDisValid() which itself also calls other functions... this is not used in the getSubjectID() in the prod file. So we need to investigate this further.

Comment (spaiva) [inline uploadNeuroDB/NeuroDB/MRI.pm]: I agree. I will look into this.

**Comment by MounaSafiHarab** [comment]
@spaiva @cmadjar 

I think you need to update also uploadNeuroDB/registerFile.pl because there is a call to getSubjectID().

**Comment by spaiva** [comment]
@MounaSafiHarab Sounds good. Thanks!

**Comment by MounaSafiHarab** [comment]
Move to 17.0

**Comment by MounaSafiHarab** [comment]
@spaiva @cmadjar 
One more thing that occurred to me now, the pipeline calls IsCandidateInfoValid() which greps the DICOM header for the PatientName field in cases of non-phantom entries to validate this field being PSCID_DCCID_VisitLabel, but since now the study can choose PatientName or PatientID as the identifier, then this function should also be made configurable as to which DICOM header to validate.

**Comment by MounaSafiHarab** [comment]
@spaiva 
I will set the milestone to 17.1 to be in sync with redmine I guess.

**Comment by spaiva** [comment]
Ready for testing @MounaSafiHarab and @cmadjar. Thanks so much for your time.

**Comment by MounaSafiHarab** [comment]
I added the needs work label just to make sure that the changes are still valid and do not conflict with other requests that have been merged, since it has been a while (I am no longer sure which release this is going to)

**Pull request closed without merging**